raft/testdata/campaign_learner_must_vote.txt

```diff
@@ -0,0 +1,123 @@
+# Regression test that verifies that learners can vote. This holds only in the
+# sense that if a learner is asked to vote, a candidate believes that they are a
+# voter based on its current config, which may be more recent than that of the
+# learner. If learners which are actually voters but don't know it yet don't
+# vote in that situation, the raft group may end up unavailable despite a quorum
+# of voters (as of the latest config) being available.
+#
+# See:
+# https://github.com/etcd-io/etcd/pull/10998
+
+# Turn output off during boilerplate.
+log-level none
+----
+ok
+
+add-nodes 3 voters=(1,2) learners=(3) index=2
+----
+ok
+
+campaign 1
+----
+ok
+
+stabilize
+----
+ok (quiet)
+
+propose-conf-change 1
+v3
+----
+ok
+
+stabilize 1 2
+----
+ok (quiet)
+
+log-level debug
+----
+ok
+
+campaign 2
+----
+INFO 2 is starting a new election at term 1
+INFO 2 became candidate at term 2
+INFO 2 received MsgVoteResp from 2 at term 2
+INFO 2 [logterm: 1, index: 4] sent MsgVote request to 1 at term 2
+INFO 2 [logterm: 1, index: 4] sent MsgVote request to 3 at term 2
+
+# n2 is now campaigning while n1 is down (does not respond). The latest config
+# has n1 as a voter, but n1 doesn't even have the corresponding conf change in
```

Comment:
> but n1 doesn't even have the corresponding conf change in its log

Doesn't it? Don't we see that with:

```
1->3 MsgApp Term:1 Log:1/3 Commit:3 Entries:[1/4 EntryConfChangeV2 v3]
1->3 MsgApp Term:1 Log:1/4 Commit:4
2->3 MsgVote Term:2 Log:1/4
```

Reply:
🙈 I needed to explicitly drop the messages before. Fixed now along with some more cleanup, and I restructured the test so that we see the vote response clearly in its own block.